features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/slowframes/DefaultSlowFramesListenerTest.kt

```diff
@@ -0,0 +1,318 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+package com.datadog.android.rum.internal.metric.slowframes
+
+import androidx.metrics.performance.FrameData
+import com.datadog.android.rum.internal.domain.state.SlowFrameRecord
+import com.datadog.android.rum.utils.forge.Configurator
+import com.datadog.tools.unit.extensions.TestConfigurationExtension
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.LongForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(TestConfigurationExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = Configurator::class)
+internal class DefaultSlowFramesListenerTest {
+
+    @StringForgery
+    lateinit var viewId: String
+
+    @LongForgery(min = 1L, Long.MAX_VALUE / 3)
+    var viewCreatedTimestampNs: Long = 0L
+
+    private lateinit var testedListener: DefaultSlowFramesListener
+
+    @BeforeEach
+    fun `set up`() {
+        testedListener = DefaultSlowFramesListener(
+            frozenFrameThresholdNs = Long.MAX_VALUE
+        )
+    }
+
+    @Test
+    fun `M return non empty report W resolveReport { jank frame occurred }`(forge: Forge) {
+        // Given
+        testedListener.onViewCreated(viewId, viewCreatedTimestampNs)
+        val jankFrameData = forge.aFrameData()
+        testedListener.onFrame(jankFrameData)
+
+        // When
+        val report = testedListener.resolveReport(viewId)
+
+        // Then
+        assertThat(report.slowFramesRecords.size).isEqualTo(1)
```

Comment:
is `hasSize` available here?

```suggestion
        assertThat(report.slowFramesRecords).hasSize(1)
```